### PR TITLE
Db utlities

### PR DESCRIPTION
**Pull request by jordan-carson**
Updated db_utilities to include several new functions, mainly to bulk_insert data into a database. There are helper functions to prepare the query, and to prepare the bulk inserts into chunks. We also check whether the schema.table exists already.

### CODE REVIEW

**Comment by matthewmturner** [comment]
this has merge conflict that must be resolved - did you check it out?

File: rice/db_utilities/db_access.py [hunk 1]
```diff
@@ -1,92 +1,104 @@
...
+
+        for field in required_fields:
+            if not getattr(self, field):
+                raise ValueError(f"Field={field} must hold a value")
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} Server={self.hostname}:{self.port}, Database={self.database}>"
+
+
+DATASOURCES = {
```
Comment (matthewmturner) [inline rice/db_utilities/db_access.py:55]: DB credentials shouldnt be visible in code. needs to be handled locally. dont want my IP address to be public.

**Comment by matthewmturner** [comment]
is this what youre currently working on?

**Comment by jordan-carson** [comment]
> is this what youre currently working on?

Yes, I fixed the issues and was able to connect to postgresql using the db_utilities file

**Comment by jordan-carson** [comment]
Tests have passed - lets move this to master

**Pull request merged**